entities/src/main/java/com/flipkart/varadhi/entities/config/MessageHeaderConfiguration.java

```diff
@@ -0,0 +1,75 @@
+package com.flipkart.varadhi.entities.config;
+
+import com.flipkart.varadhi.entities.Validatable;
+import com.flipkart.varadhi.entities.constants.StandardHeaders;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+
+import java.util.List;
+import java.util.Map;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@Data
+@Builder
+public class MessageHeaderConfiguration implements Validatable {
+    @NotNull
+    private final Map<StandardHeaders, String> mapping;
+    @NotNull
+    private final List<String> allowedPrefix;
+    @NotNull
+    private final Integer headerValueSizeMax;
+    @NotNull
+    private final Integer maxRequestSize;
+    @JsonCreator
+    public MessageHeaderConfiguration(
```

Comment:
Do we really need this constructor. any reason we are not chosing auto mapping of configs?